### PR TITLE
Issue #11747: Migrate codenarc to travis CI

### LINKED ISSUE #11747: Move codenarc to non GitHub CI
From https://github.com/checkstyle/checkstyle/pull/11739#issuecomment-1157245833

I would move this check to Travis or other not high demand CI. It it most likely that such groovy script(s) will never be updated after initial creation, so better to have it run in some less critical CI.

### PR DESCRIPTION
**Pull request by Vyom-Yadav**
Resolves #11747 

### Proof of working:
Failing due to extra import - https://app.travis-ci.com/github/checkstyle/checkstyle/jobs/586922249

### CODE REVIEW

File: .ci/codenarc.sh [hunk 1]
```diff
@@ -1,6 +1,5 @@
 #!/bin/bash
 # Attention, there is no "-x" to avoid problems on Travis
-set -e
```
Comment (Vyom-Yadav) [inline .ci/codenarc.sh:3]: No need to exit if `groovy ./.ci/codenarc.groovy` fails, we still need to echo the reason for failure, for example, if groovy is not installed or `JAVA_HOME` is not set.

Comment (romani) [reply .ci/codenarc.sh:3]: This is required to fail fast and do not continue execution I'd something existing with error code.

Please share more details and log on this update

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: .ci/codenarc.sh (hunk 1 above)
Comment (Vyom-Yadav) [reply .ci/codenarc.sh]: Actually `set -e` can stay, error in groovy installation or `JAVA_HOME` would be shown in `before_script`

Comment (romani) [reply .ci/codenarc.sh]: Yes, placing `-e` in Travis level code is damaging it's execution, but in script it is ok

**Review by romani — APPROVED** [review]
Ok to merge

File: .travis.yml [hunk 2]
```diff
@@ -43,13 +44,26 @@ jobs:
...
     - env:
         - DESC="ensure that all modules are used in no exception configs"
         - CMD1="export PULL_REQUEST=$TRAVIS_PULL_REQUEST"
         - CMD2="./.ci/validation.sh verify-no-exception-configs"
         - CMD="$CMD1 && $CMD2"
 
+before_script:
+  - |
+    if [[ $TRAVIS_CPU_ARCH == 'ppc64le' ]]; then
+        export JAVA_HOME=/usr/lib/jvm/adoptopenjdk-11-hotspot-ppc64el
```
Comment (rnveach) [inline .travis.yml:62]: @romani Should this actually be in `./.ci/travis.sh install-adoptium-jdk` as it is the JRE and to keep it all together as needed for travis.

**Review by rnveach — APPROVED** [review]
Just the one comment.

File: .travis.yml (hunk 2 above)
Comment (nrmancuso) [reply .travis.yml:62]: Let's keep evidence of weird architecture in yml file, to remind us of this when adding new jobs here.

**Review by nrmancuso — APPROVED** [review]

**Pull request merged**